{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000102693.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000064842.1\"; gene_type \"snR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>transcript</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000064842.1\"; transcript_id ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1           2        3        4  5  6  7  \\\n",
       "0  chr1   HAVANA        gene  3073253  3074322  .  +  .   \n",
       "1  chr1   HAVANA  transcript  3073253  3074322  .  +  .   \n",
       "2  chr1   HAVANA        exon  3073253  3074322  .  +  .   \n",
       "3  chr1  ENSEMBL        gene  3102016  3102125  .  +  .   \n",
       "4  chr1  ENSEMBL  transcript  3102016  3102125  .  +  .   \n",
       "\n",
       "                                                   8  \n",
       "0  gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC...  \n",
       "1  gene_id \"ENSMUSG00000102693.1\"; transcript_id ...  \n",
       "2  gene_id \"ENSMUSG00000102693.1\"; transcript_id ...  \n",
       "3  gene_id \"ENSMUSG00000064842.1\"; gene_type \"snR...  \n",
       "4  gene_id \"ENSMUSG00000064842.1\"; transcript_id ...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtf = pd.read_csv('gencode.vM21.annotation.gtf.gz', sep='\\t', comment='#', header=None)\n",
    "df_gtf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          OTTMUSG00000049935.1\n",
       "1                 4933401J01Rik\n",
       "2                 4933401J01Rik\n",
       "3                           NaN\n",
       "4                       Gm26206\n",
       "5                       Gm26206\n",
       "6          OTTMUSG00000026353.2\n",
       "7                          Xkr4\n",
       "8                          Xkr4\n",
       "9                          Xkr4\n",
       "10                         Xkr4\n",
       "11                         Xkr4\n",
       "12                         Xkr4\n",
       "13                         Xkr4\n",
       "14                         Xkr4\n",
       "15                         Xkr4\n",
       "16                         Xkr4\n",
       "17                         Xkr4\n",
       "18                         Xkr4\n",
       "19                         Xkr4\n",
       "20                         Xkr4\n",
       "21                         Xkr4\n",
       "22                         Xkr4\n",
       "23                         Xkr4\n",
       "24               pseudo_consens\n",
       "25                      Gm18956\n",
       "26                      Gm18956\n",
       "27         OTTMUSG00000049960.1\n",
       "28                      Gm37180\n",
       "29                      Gm37180\n",
       "                   ...         \n",
       "1852160                     NaN\n",
       "1852161                  mt-Tl2\n",
       "1852162                  mt-Tl2\n",
       "1852163                     NaN\n",
       "1852164                  mt-Nd5\n",
       "1852165                  mt-Nd5\n",
       "1852166                  mt-Nd5\n",
       "1852167                  mt-Nd5\n",
       "1852168                  mt-Nd5\n",
       "1852169                     NaN\n",
       "1852170                  mt-Nd6\n",
       "1852171                  mt-Nd6\n",
       "1852172                  mt-Nd6\n",
       "1852173                  mt-Nd6\n",
       "1852174                  mt-Nd6\n",
       "1852175                  mt-Nd6\n",
       "1852176                     NaN\n",
       "1852177                   mt-Te\n",
       "1852178                   mt-Te\n",
       "1852179                     NaN\n",
       "1852180                 mt-Cytb\n",
       "1852181                 mt-Cytb\n",
       "1852182                 mt-Cytb\n",
       "1852183                 mt-Cytb\n",
       "1852184                     NaN\n",
       "1852185                   mt-Tt\n",
       "1852186                   mt-Tt\n",
       "1852187                     NaN\n",
       "1852188                   mt-Tp\n",
       "1852189                   mt-Tp\n",
       "Name: 8, Length: 1852190, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtf[8].str.split('\"').str.get(7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1775744</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>7573644</td>\n",
       "      <td>7595245</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; gene_type \"pr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775745</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>7573644</td>\n",
       "      <td>7586518</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775746</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7573644</td>\n",
       "      <td>7573909</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775747</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7577688</td>\n",
       "      <td>7577724</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775748</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7577908</td>\n",
       "      <td>7578081</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775749</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7586334</td>\n",
       "      <td>7586518</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775750</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586518</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775751</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586357</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775752</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7573644</td>\n",
       "      <td>7573909</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775753</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7577688</td>\n",
       "      <td>7577724</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775754</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7577908</td>\n",
       "      <td>7578081</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775755</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7586334</td>\n",
       "      <td>7586354</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775756</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>7579567</td>\n",
       "      <td>7593799</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775757</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7579567</td>\n",
       "      <td>7579725</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775758</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7586334</td>\n",
       "      <td>7586561</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775759</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586561</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775760</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586357</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775761</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587074</td>\n",
       "      <td>7587178</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775762</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587074</td>\n",
       "      <td>7587178</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775763</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587275</td>\n",
       "      <td>7587413</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775764</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587275</td>\n",
       "      <td>7587413</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775765</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587787</td>\n",
       "      <td>7587874</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775766</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587787</td>\n",
       "      <td>7587874</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775767</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587973</td>\n",
       "      <td>7588077</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775768</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587973</td>\n",
       "      <td>7588077</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775769</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7588868</td>\n",
       "      <td>7588955</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775770</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7588868</td>\n",
       "      <td>7588955</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775771</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7589096</td>\n",
       "      <td>7589176</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775772</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7589096</td>\n",
       "      <td>7589176</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775773</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7591373</td>\n",
       "      <td>7591526</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775986</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7592650</td>\n",
       "      <td>7592751</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775987</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>exon</td>\n",
       "      <td>7592900</td>\n",
       "      <td>7595243</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775988</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7592900</td>\n",
       "      <td>7593040</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775989</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>stop_codon</td>\n",
       "      <td>7593041</td>\n",
       "      <td>7593043</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775990</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7586273</td>\n",
       "      <td>7586354</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775991</th>\n",
       "      <td>chrX</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7593041</td>\n",
       "      <td>7595243</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775992</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>transcript</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7593048</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775993</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586561</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775994</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586561</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775995</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>start_codon</td>\n",
       "      <td>7586355</td>\n",
       "      <td>7586357</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775996</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587074</td>\n",
       "      <td>7587178</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775997</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587074</td>\n",
       "      <td>7587178</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775998</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587275</td>\n",
       "      <td>7587413</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775999</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587275</td>\n",
       "      <td>7587413</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776000</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587787</td>\n",
       "      <td>7587874</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776001</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587787</td>\n",
       "      <td>7587874</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776002</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7587973</td>\n",
       "      <td>7588077</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776003</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7587973</td>\n",
       "      <td>7588077</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776004</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7588868</td>\n",
       "      <td>7588955</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776005</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7588868</td>\n",
       "      <td>7588955</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>1</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776006</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7591373</td>\n",
       "      <td>7591526</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776007</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7591373</td>\n",
       "      <td>7591526</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776008</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7592159</td>\n",
       "      <td>7592235</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776009</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7592159</td>\n",
       "      <td>7592235</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>2</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776010</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7592650</td>\n",
       "      <td>7592751</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776011</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7592650</td>\n",
       "      <td>7592751</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776012</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>exon</td>\n",
       "      <td>7592900</td>\n",
       "      <td>7593048</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776013</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>CDS</td>\n",
       "      <td>7592900</td>\n",
       "      <td>7593040</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776014</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>stop_codon</td>\n",
       "      <td>7593041</td>\n",
       "      <td>7593043</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776015</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>UTR</td>\n",
       "      <td>7593041</td>\n",
       "      <td>7593048</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; transcript_id...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>272 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            0        1            2        3        4  5  6  7  \\\n",
       "1775744  chrX   HAVANA         gene  7573644  7595245  .  +  .   \n",
       "1775745  chrX   HAVANA   transcript  7573644  7586518  .  +  .   \n",
       "1775746  chrX   HAVANA         exon  7573644  7573909  .  +  .   \n",
       "1775747  chrX   HAVANA         exon  7577688  7577724  .  +  .   \n",
       "1775748  chrX   HAVANA         exon  7577908  7578081  .  +  .   \n",
       "1775749  chrX   HAVANA         exon  7586334  7586518  .  +  .   \n",
       "1775750  chrX   HAVANA          CDS  7586355  7586518  .  +  0   \n",
       "1775751  chrX   HAVANA  start_codon  7586355  7586357  .  +  0   \n",
       "1775752  chrX   HAVANA          UTR  7573644  7573909  .  +  .   \n",
       "1775753  chrX   HAVANA          UTR  7577688  7577724  .  +  .   \n",
       "1775754  chrX   HAVANA          UTR  7577908  7578081  .  +  .   \n",
       "1775755  chrX   HAVANA          UTR  7586334  7586354  .  +  .   \n",
       "1775756  chrX   HAVANA   transcript  7579567  7593799  .  +  .   \n",
       "1775757  chrX   HAVANA         exon  7579567  7579725  .  +  .   \n",
       "1775758  chrX   HAVANA         exon  7586334  7586561  .  +  .   \n",
       "1775759  chrX   HAVANA          CDS  7586355  7586561  .  +  0   \n",
       "1775760  chrX   HAVANA  start_codon  7586355  7586357  .  +  0   \n",
       "1775761  chrX   HAVANA         exon  7587074  7587178  .  +  .   \n",
       "1775762  chrX   HAVANA          CDS  7587074  7587178  .  +  0   \n",
       "1775763  chrX   HAVANA         exon  7587275  7587413  .  +  .   \n",
       "1775764  chrX   HAVANA          CDS  7587275  7587413  .  +  0   \n",
       "1775765  chrX   HAVANA         exon  7587787  7587874  .  +  .   \n",
       "1775766  chrX   HAVANA          CDS  7587787  7587874  .  +  2   \n",
       "1775767  chrX   HAVANA         exon  7587973  7588077  .  +  .   \n",
       "1775768  chrX   HAVANA          CDS  7587973  7588077  .  +  1   \n",
       "1775769  chrX   HAVANA         exon  7588868  7588955  .  +  .   \n",
       "1775770  chrX   HAVANA          CDS  7588868  7588955  .  +  1   \n",
       "1775771  chrX   HAVANA         exon  7589096  7589176  .  +  .   \n",
       "1775772  chrX   HAVANA          CDS  7589096  7589176  .  +  0   \n",
       "1775773  chrX   HAVANA         exon  7591373  7591526  .  +  .   \n",
       "...       ...      ...          ...      ...      ... .. .. ..   \n",
       "1775986  chrX  ENSEMBL          CDS  7592650  7592751  .  +  0   \n",
       "1775987  chrX  ENSEMBL         exon  7592900  7595243  .  +  .   \n",
       "1775988  chrX  ENSEMBL          CDS  7592900  7593040  .  +  0   \n",
       "1775989  chrX  ENSEMBL   stop_codon  7593041  7593043  .  +  0   \n",
       "1775990  chrX  ENSEMBL          UTR  7586273  7586354  .  +  .   \n",
       "1775991  chrX  ENSEMBL          UTR  7593041  7595243  .  +  .   \n",
       "1775992  chrX   HAVANA   transcript  7586355  7593048  .  +  .   \n",
       "1775993  chrX   HAVANA         exon  7586355  7586561  .  +  .   \n",
       "1775994  chrX   HAVANA          CDS  7586355  7586561  .  +  0   \n",
       "1775995  chrX   HAVANA  start_codon  7586355  7586357  .  +  0   \n",
       "1775996  chrX   HAVANA         exon  7587074  7587178  .  +  .   \n",
       "1775997  chrX   HAVANA          CDS  7587074  7587178  .  +  0   \n",
       "1775998  chrX   HAVANA         exon  7587275  7587413  .  +  .   \n",
       "1775999  chrX   HAVANA          CDS  7587275  7587413  .  +  0   \n",
       "1776000  chrX   HAVANA         exon  7587787  7587874  .  +  .   \n",
       "1776001  chrX   HAVANA          CDS  7587787  7587874  .  +  2   \n",
       "1776002  chrX   HAVANA         exon  7587973  7588077  .  +  .   \n",
       "1776003  chrX   HAVANA          CDS  7587973  7588077  .  +  1   \n",
       "1776004  chrX   HAVANA         exon  7588868  7588955  .  +  .   \n",
       "1776005  chrX   HAVANA          CDS  7588868  7588955  .  +  1   \n",
       "1776006  chrX   HAVANA         exon  7591373  7591526  .  +  .   \n",
       "1776007  chrX   HAVANA          CDS  7591373  7591526  .  +  0   \n",
       "1776008  chrX   HAVANA         exon  7592159  7592235  .  +  .   \n",
       "1776009  chrX   HAVANA          CDS  7592159  7592235  .  +  2   \n",
       "1776010  chrX   HAVANA         exon  7592650  7592751  .  +  .   \n",
       "1776011  chrX   HAVANA          CDS  7592650  7592751  .  +  0   \n",
       "1776012  chrX   HAVANA         exon  7592900  7593048  .  +  .   \n",
       "1776013  chrX   HAVANA          CDS  7592900  7593040  .  +  0   \n",
       "1776014  chrX   HAVANA   stop_codon  7593041  7593043  .  +  0   \n",
       "1776015  chrX   HAVANA          UTR  7593041  7593048  .  +  .   \n",
       "\n",
       "                                                         8  \n",
       "1775744  gene_id \"ENSMUSG00000039521.13\"; gene_type \"pr...  \n",
       "1775745  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775746  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775747  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775748  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775749  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775750  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775751  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775752  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775753  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775754  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775755  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775756  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775757  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775758  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775759  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775760  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775761  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775762  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775763  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775764  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775765  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775766  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775767  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775768  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775769  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775770  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775771  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775772  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775773  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "...                                                    ...  \n",
       "1775986  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775987  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775988  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775989  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775990  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775991  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775992  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775993  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775994  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775995  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775996  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775997  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775998  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1775999  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776000  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776001  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776002  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776003  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776004  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776005  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776006  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776007  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776008  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776009  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776010  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776011  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776012  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776013  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776014  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "1776015  gene_id \"ENSMUSG00000039521.13\"; transcript_id...  \n",
       "\n",
       "[272 rows x 9 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_foxp3 = df_gtf[['Foxp3' in x for x in df_gtf[8]]]\n",
    "df_foxp3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1775744</th>\n",
       "      <td>chrX</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>7573644</td>\n",
       "      <td>7595245</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>gene_id \"ENSMUSG00000039521.13\"; gene_type \"pr...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            0       1     2        3        4  5  6  7  \\\n",
       "1775744  chrX  HAVANA  gene  7573644  7595245  .  +  .   \n",
       "\n",
       "                                                         8  \n",
       "1775744  gene_id \"ENSMUSG00000039521.13\"; gene_type \"pr...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_foxp3[df_foxp3[2] == 'gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7573644"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_foxp3[3].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7595245"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_foxp3[4].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC...\n",
       "3          gene_id \"ENSMUSG00000064842.1\"; gene_type \"snR...\n",
       "6          gene_id \"ENSMUSG00000051951.5\"; gene_type \"pro...\n",
       "24         gene_id \"ENSMUSG00000102851.1\"; gene_type \"pro...\n",
       "27         gene_id \"ENSMUSG00000103377.1\"; gene_type \"TEC...\n",
       "30         gene_id \"ENSMUSG00000104017.1\"; gene_type \"TEC...\n",
       "33         gene_id \"ENSMUSG00000103025.1\"; gene_type \"TEC...\n",
       "36         gene_id \"ENSMUSG00000089699.1\"; gene_type \"ant...\n",
       "40         gene_id \"ENSMUSG00000103201.1\"; gene_type \"TEC...\n",
       "43         gene_id \"ENSMUSG00000103147.1\"; gene_type \"pro...\n",
       "46         gene_id \"ENSMUSG00000103161.1\"; gene_type \"TEC...\n",
       "49         gene_id \"ENSMUSG00000102331.1\"; gene_type \"sen...\n",
       "53         gene_id \"ENSMUSG00000102348.1\"; gene_type \"TEC...\n",
       "56         gene_id \"ENSMUSG00000102592.1\"; gene_type \"TEC...\n",
       "59         gene_id \"ENSMUSG00000088333.2\"; gene_type \"snR...\n",
       "62         gene_id \"ENSMUSG00000102343.1\"; gene_type \"lin...\n",
       "70         gene_id \"ENSMUSG00000025900.12\"; gene_type \"pr...\n",
       "166        gene_id \"ENSMUSG00000102948.1\"; gene_type \"pro...\n",
       "170        gene_id \"ENSMUSG00000104123.1\"; gene_type \"TEC...\n",
       "173        gene_id \"ENSMUSG00000025902.13\"; gene_type \"pr...\n",
       "257        gene_id \"ENSMUSG00000104238.1\"; gene_type \"pro...\n",
       "264        gene_id \"ENSMUSG00000102269.1\"; gene_type \"pro...\n",
       "268        gene_id \"ENSMUSG00000096126.1\"; gene_type \"snR...\n",
       "271        gene_id \"ENSMUSG00000103003.1\"; gene_type \"pro...\n",
       "274        gene_id \"ENSMUSG00000104328.1\"; gene_type \"lin...\n",
       "278        gene_id \"ENSMUSG00000102735.1\"; gene_type \"pro...\n",
       "281        gene_id \"ENSMUSG00000098104.1\"; gene_type \"pro...\n",
       "284        gene_id \"ENSMUSG00000102175.1\"; gene_type \"pro...\n",
       "287        gene_id \"ENSMUSG00000088000.1\"; gene_type \"snR...\n",
       "290        gene_id \"ENSMUSG00000103265.1\"; gene_type \"pro...\n",
       "                                 ...                        \n",
       "1852061    gene_id \"ENSMUSG00000064343.1\"; gene_type \"Mt_...\n",
       "1852064    gene_id \"ENSMUSG00000064344.1\"; gene_type \"Mt_...\n",
       "1852067    gene_id \"ENSMUSG00000064345.1\"; gene_type \"pro...\n",
       "1852073    gene_id \"ENSMUSG00000064346.1\"; gene_type \"Mt_...\n",
       "1852076    gene_id \"ENSMUSG00000064347.1\"; gene_type \"Mt_...\n",
       "1852079    gene_id \"ENSMUSG00000064348.1\"; gene_type \"Mt_...\n",
       "1852082    gene_id \"ENSMUSG00000064349.1\"; gene_type \"Mt_...\n",
       "1852085    gene_id \"ENSMUSG00000064350.1\"; gene_type \"Mt_...\n",
       "1852088    gene_id \"ENSMUSG00000064351.1\"; gene_type \"pro...\n",
       "1852095    gene_id \"ENSMUSG00000064352.1\"; gene_type \"Mt_...\n",
       "1852098    gene_id \"ENSMUSG00000064353.1\"; gene_type \"Mt_...\n",
       "1852101    gene_id \"ENSMUSG00000064354.1\"; gene_type \"pro...\n",
       "1852108    gene_id \"ENSMUSG00000064355.1\"; gene_type \"Mt_...\n",
       "1852111    gene_id \"ENSMUSG00000064356.3\"; gene_type \"pro...\n",
       "1852118    gene_id \"ENSMUSG00000064357.1\"; gene_type \"pro...\n",
       "1852125    gene_id \"ENSMUSG00000064358.1\"; gene_type \"pro...\n",
       "1852130    gene_id \"ENSMUSG00000064359.1\"; gene_type \"Mt_...\n",
       "1852133    gene_id \"ENSMUSG00000064360.1\"; gene_type \"pro...\n",
       "1852139    gene_id \"ENSMUSG00000064361.1\"; gene_type \"Mt_...\n",
       "1852142    gene_id \"ENSMUSG00000065947.3\"; gene_type \"pro...\n",
       "1852149    gene_id \"ENSMUSG00000064363.1\"; gene_type \"pro...\n",
       "1852154    gene_id \"ENSMUSG00000064364.1\"; gene_type \"Mt_...\n",
       "1852157    gene_id \"ENSMUSG00000064365.1\"; gene_type \"Mt_...\n",
       "1852160    gene_id \"ENSMUSG00000064366.1\"; gene_type \"Mt_...\n",
       "1852163    gene_id \"ENSMUSG00000064367.1\"; gene_type \"pro...\n",
       "1852169    gene_id \"ENSMUSG00000064368.1\"; gene_type \"pro...\n",
       "1852176    gene_id \"ENSMUSG00000064369.1\"; gene_type \"Mt_...\n",
       "1852179    gene_id \"ENSMUSG00000064370.1\"; gene_type \"pro...\n",
       "1852184    gene_id \"ENSMUSG00000064371.1\"; gene_type \"Mt_...\n",
       "1852187    gene_id \"ENSMUSG00000064372.1\"; gene_type \"Mt_...\n",
       "Name: 8, Length: 55450, dtype: object"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtf[df_gtf[2] == 'gene'][8]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC\"; gene_name \"4933401J01Rik\"; level 2; havana_gene \"OTTMUSG00000049935.1\";'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s = df_gtf.loc[0, 8]\n",
    "s"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_name(s):\n",
    "    l = [x.split() for x in s.split(';')[:-1]]\n",
    "    return [x[1].replace('\"', '') for x in l if x[0] == 'gene_name'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4933401J01Rik'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_gene_name(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 8.78 s, sys: 24.3 ms, total: 8.8 s\n",
      "Wall time: 8.81 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "df_gtf['gene_name'] = df_gtf.loc[:,8].map(get_gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_gtf_gene = df_gtf[df_gtf[2] == 'gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtf_gene.loc[:, 'gene_name'].is_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9743              Ptp4a1\n",
       "12353            Arhgef4\n",
       "17253            Gm24826\n",
       "31372            Gm26457\n",
       "57493              Sept2\n",
       "61840          Zfp813-ps\n",
       "66102            Gm23370\n",
       "71952            Gm28040\n",
       "72420            Zc3h11a\n",
       "79745            Gm28724\n",
       "89806            Snord80\n",
       "95367            Gm16701\n",
       "130175             Ndor1\n",
       "135417           Snora43\n",
       "150697              Nron\n",
       "157597           Gm24350\n",
       "184666      Olfr1073-ps1\n",
       "201020           Gm22813\n",
       "205069     A530058N18Rik\n",
       "222706            Gm4430\n",
       "232520          Nkx2-2os\n",
       "245805           Gm23925\n",
       "268431            Gm2464\n",
       "273586     1600017P15Rik\n",
       "290337            Gm7270\n",
       "290587           Gm20690\n",
       "301488           Gm25820\n",
       "309733           Gm18433\n",
       "368253              Rmrp\n",
       "376423           Gm25053\n",
       "               ...      \n",
       "1299439         Ighv1-13\n",
       "1312366         Vmn1r216\n",
       "1358620          Gm36638\n",
       "1363100           Gm9025\n",
       "1365958              Nnt\n",
       "1405465      Vmn2r-ps111\n",
       "1408738           Gm6740\n",
       "1439851           Gm5089\n",
       "1440398    4930594M22Rik\n",
       "1469599            Mirt2\n",
       "1477058             Gcat\n",
       "1484755          Gm27825\n",
       "1484776          Gm28023\n",
       "1495260          Gm41392\n",
       "1517638          Gm38619\n",
       "1546809          Gm25617\n",
       "1565104            Atp5o\n",
       "1569909            Igsf5\n",
       "1573132          Gm29719\n",
       "1580773           Gm5966\n",
       "1600021           Ggnbp1\n",
       "1634504          Gm44501\n",
       "1653943           Myl12b\n",
       "1682296          Mir1949\n",
       "1685715          Pcdha11\n",
       "1688419         Arhgap26\n",
       "1707874            Hdhd2\n",
       "1715796          Gm23927\n",
       "1723500           Sssca1\n",
       "1835176          Gm23786\n",
       "Name: gene_name, Length: 109, dtype: object"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gtf_gene[df_gtf_gene.loc[:, 'gene_name'].duplicated()]['gene_name']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "l = []\n",
    "for g in df_gtf.gene_name.unique()[:10]:\n",
    "#     print(g)\n",
    "    df_q = df_gtf[df_gtf.gene_name==g]\n",
    "    chrom = list(df_q[0])[0]\n",
    "    start = df_q[3].min()\n",
    "    end = df_q[4].max()\n",
    "    strand = list(df_q[6])[0]\n",
    "    l.append([chrom, start, end, g, 0, strand])\n",
    "    \n",
    "df_bed = pd.DataFrame(l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>Gm26206</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3671498</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3252757</td>\n",
       "      <td>3253236</td>\n",
       "      <td>Gm18956</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3365731</td>\n",
       "      <td>3368549</td>\n",
       "      <td>Gm37180</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3375556</td>\n",
       "      <td>3377788</td>\n",
       "      <td>Gm37363</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3464977</td>\n",
       "      <td>3467285</td>\n",
       "      <td>Gm37686</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3466587</td>\n",
       "      <td>3513553</td>\n",
       "      <td>Gm1992</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3512451</td>\n",
       "      <td>3514507</td>\n",
       "      <td>Gm37329</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3531795</td>\n",
       "      <td>3532720</td>\n",
       "      <td>Gm7341</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1        2              3  4  5\n",
       "0  chr1  3073253  3074322  4933401J01Rik  0  +\n",
       "1  chr1  3102016  3102125        Gm26206  0  +\n",
       "2  chr1  3205901  3671498           Xkr4  0  -\n",
       "3  chr1  3252757  3253236        Gm18956  0  +\n",
       "4  chr1  3365731  3368549        Gm37180  0  -\n",
       "5  chr1  3375556  3377788        Gm37363  0  -\n",
       "6  chr1  3464977  3467285        Gm37686  0  -\n",
       "7  chr1  3466587  3513553         Gm1992  0  +\n",
       "8  chr1  3512451  3514507        Gm37329  0  -\n",
       "9  chr1  3531795  3532720         Gm7341  0  +"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3531795"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(df_q[3])[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43    3531795\n",
       "44    3531795\n",
       "45    3531795\n",
       "Name: 3, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_q[3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3531795"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_q[3].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "list_genes = ['Xkr4']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>Gm26206</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3671498</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3252757</td>\n",
       "      <td>3253236</td>\n",
       "      <td>Gm18956</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3365731</td>\n",
       "      <td>3368549</td>\n",
       "      <td>Gm37180</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3375556</td>\n",
       "      <td>3377788</td>\n",
       "      <td>Gm37363</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3464977</td>\n",
       "      <td>3467285</td>\n",
       "      <td>Gm37686</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3466587</td>\n",
       "      <td>3513553</td>\n",
       "      <td>Gm1992</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3512451</td>\n",
       "      <td>3514507</td>\n",
       "      <td>Gm37329</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3531795</td>\n",
       "      <td>3532720</td>\n",
       "      <td>Gm7341</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1        2              3  4  5\n",
       "0  chr1  3073253  3074322  4933401J01Rik  0  +\n",
       "1  chr1  3102016  3102125        Gm26206  0  +\n",
       "2  chr1  3205901  3671498           Xkr4  0  -\n",
       "3  chr1  3252757  3253236        Gm18956  0  +\n",
       "4  chr1  3365731  3368549        Gm37180  0  -\n",
       "5  chr1  3375556  3377788        Gm37363  0  -\n",
       "6  chr1  3464977  3467285        Gm37686  0  -\n",
       "7  chr1  3466587  3513553         Gm1992  0  +\n",
       "8  chr1  3512451  3514507        Gm37329  0  -\n",
       "9  chr1  3531795  3532720         Gm7341  0  +"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_q = df_bed[df_bed[3].isin(list_genes)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3671498</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1        2     3  4  5\n",
       "2  chr1  3205901  3671498  Xkr4  0  -"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "mergin = 50000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/yasumizuyoshiaki/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n",
      "/Users/yasumizuyoshiaki/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "df_q[1] = df_q[1] - mergin\n",
    "df_q[2] = df_q[2] + mergin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3155901</td>\n",
       "      <td>3721498</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      0        1        2     3  4  5\n",
       "2  chr1  3155901  3721498  Xkr4  0  -"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_q"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
